source/common/network/cidr_range.cc

```diff
@@ -79,13 +79,20 @@ IpVersion CidrRange::version() const {
 }
 
 bool CidrRange::isInRange(InstanceConstSharedPtr address) const {
-  if (address == nullptr || length_ < 0 || address->type() != Type::Ip ||
-      address->ip()->version() != version()) {
+  if (address == nullptr) {
+    return false;
+  }
+
+  return isInRange(*address);
+}
+
+bool CidrRange::isInRange(const Instance& address) const {
+  if (length_ < 0 || address.type() != Type::Ip || address.ip()->version() != version()) {
     return false;
   }
   // Make an CidrRange from the address, of the same length as this. If the two ranges have
   // are the same, then the address is in this range.
-  CidrRange other = create(address, length_);
+  CidrRange other = create(address.ip()->addressAsString(), length_);
```

Comment:
I think the underlying problem is that CidrRange consists of an InstanceConstSharedPtr. Eventually the conversion from the Instance object to shared pointer must be made. It can happen in create, truncateIpAddressAndLength, or in the CidrRange constructor.